coriolis_openstack_utils/actions.py

```diff
@@ -753,3 +755,157 @@ def execute_operations(self):
         # done_ids = [migr["existing_migration_id"]
         #             for migr in self._completed_migrations]
         return migrations
+
+
+class SecurityGroupCreationAction(Action):
+    action_type = ACTION_TYPE_CHECK_CREATE_SECGROUP
+    NEW_SECGROUP_DESCRIPTION = (
+        "Created by the Coriolis OpenStack utilities for source security group"
+        " '%s'.")
+
+    def __init__(self, source_client, destination_client, action_payload):
+        """
+        param source_client: source client needed for fetching source
+        secgroup info
+        param destination_client: destination client needed for secgroup
+        creation
+        param action_payload: dict(): payload (params) for the action
+        must contain 'src_tenant_id' or 'src_project_id'
+                     'dest_tenant_id' / 'dest_tenant_id'
+                     'source_name'
+        """
+        self._source_client = source_client
+        self._destination_client = destination_client
+        self.payload = action_payload
+        self.subactions = []
+
+    @property
+    def secgroup_name_format(self):
+        return CONF.destination.new_secgroup_name_format
+
+    def get_source_tenant_id(self):
+        return (self.payload.get('src_tenant_id') or
+                self.payload.get('src_project_id'))
+
+    def get_destination_tenant_id(self):
+        return (self.payload.get('dest_tenant_id') or
+                self.payload.get('dest_project_id'))
+
+    def check_already_done(self):
+        dest_tenant_id = self.get_destination_tenant_id()
+        secgroup_name = self.get_new_secgroup_name()
+
+        conflicting = len(security_groups.list_security_groups(
+            self._destination_client,
+            dest_tenant_id,
+            secgroup_name=secgroup_name))
+
+        if conflicting == 1:
+            return {"done": True, "result": secgroup_name}
+        elif conflicting >= 2:
```

Comment:
Why would finding one secgroup named that way be ok but multiples lead to an error?
Anyhow, secgroup names must be unique *within a given tenant* **afaik** (please double-check this as it may be different using the Neutron secgrups API directly vs. the Nova ones!), so we should only ever find one, so this extra check is redundant.